addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -70,8 +70,11 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             $amountUntaxed.html(result.new_amount_untaxed);
             $amountTax.html(result.new_amount_tax);
             $amountTotal.html(result.new_amount_total);
+
+
+            $payButton.prop('disabled', false);
+            $payButton.data('disabled_reasons', $payButton.data('disabled_reasons') || {});
             $payButton.data('disabled_reasons').carrier_selection = false;
-            $payButton.prop('disabled', _.contains($payButton.data('disabled_reasons'), true));
```

Comment:
leave it in place.

The fix was in reality the selector and $payButton.data('disabled_reasons', $payButton.data('disabled_reasons') || {});